### PR TITLE
gh-113317: AC converter: Use add_include() in bad_argument()

### PR DESCRIPTION
**Pull request by vstinner**
Rename also the 'clinic' global variable to 'global_clinic'.

* Issue: gh-113317

### CODE REVIEW

File: Tools/clinic/clinic.py [hunk 1]
```diff
@@ -2347,7 +2347,7 @@ def __init__(
             preset[name] = buffer
 
         global clinic
-        clinic = self
+        global_clinic = self  # noqa: F841
```
Comment (erlend-aasland) [inline Tools/clinic/clinic.py]: You should fix `global clinic` instead of masking out that bug.

Comment (vstinner) [reply Tools/clinic/clinic.py]: Oooops, right. I also spotted that failure while running tests locally.

**Comment by erlend-aasland** [comment]
> Ah, my simple CConverter.add_include() implementation reached its limits:
> 
> ```
>   File "/home/runner/work/cpython/cpython/./Tools/clinic/clinic.py", line 3266, in bad_argument
>     self.add_include('pycore_modsupport.h', '_PyArg_BadArgument()')
>   File "/home/runner/work/cpython/cpython/./Tools/clinic/clinic.py", line 3339, in add_include
>     raise ValueError("a converter only supports a single include")
> ValueError: a converter only supports a single include
> ```

Yeah, I suspected that :) Well, it should be straight forward to support multiple includes in converters; we could extract the nice "condition" logic you added for `Clinic.add_include` and reuse that for `CConverter.add_include`.

**Comment by erlend-aasland** [comment]
Also, I think the `clinic` global renaming should be kept out; this PR is likely to grow a little bit.

**Comment by vstinner** [comment]
Sorry, I should have run tests locally, or mark this PR as a draft. I didn't expect so many back and forth.

All issues should now be fixed (let's see what the CIs say).

File: Tools/clinic/clinic.py [hunk 2]
```diff
@@ -1455,6 +1449,14 @@ def parser_body(
         if compiler_warning:
             parser_definition = compiler_warning + "\n\n" + parser_definition
 
+        # Copy converters includes at the end, since includes can be
+        # added late
+        for converter in converters:
+            include = converter.include
+            if include is not None:
```
Comment (erlend-aasland) [inline Tools/clinic/clinic.py:1456]: I think that can be made an assert.

Comment (vstinner) [reply Tools/clinic/clinic.py:1456]: Not all converters need a special include. Do you mean that `include` variable is never None?

**Comment by vstinner** [comment]
commit e14930ff6397439759eb34ca70a3493baa845014 was merged instead, I close my PR.

**Pull request closed without merging**